packages/block-editor/src/hooks/gap.js

```diff
@@ -95,18 +97,28 @@ export function GapEdit( props ) {
 		],
 	} );
 
+	const sides = useCustomSides( blockName, 'blockGap' );
+	const splitOnAxis =
+		sides && sides.some( ( side ) => AXIAL_SIDES.includes( side ) );
+
 	const ref = useBlockRef( clientId );
 
 	if ( useIsGapDisabled( props ) ) {
 		return null;
 	}
 
 	const onChange = ( next ) => {
+		let newValue = next;
+		if ( typeof next === 'object' ) {
+			const row = next?.top || 0;
+			const column = next?.left || 0;
+			newValue = row === column ? row : `${ row } ${ column }`;
```

Comment:
Why are we storing the value as a concatenated string, shouldn't we store the next object as is like we do in other kind of properties (padding, margin...)? Is it because we only allow two sides? if that's the case should we instead use a `{ vertical, horizontal }` instead if `{ top, left }` is not good enough?